test/client_test.rb

```diff
@@ -51,4 +51,17 @@ def test_standard_values
     summary_metric = client.register(:summary, 'summary_metric', 'helping', expected_quantiles)
     assert_equal(expected_quantiles, summary_metric.standard_values('value', 'key')[:opts])
   end
+
+  def test_overriding_logger
+    mock_logger = Minitest::Mock.new
```

Comment:
I know it is a lot to ask, but is there any way to do this test in a less mocky way? perhaps pass our logger in, then trigger a condition that causes some logging? 

Maybe we expose max_queue_size, so we dont need to reach in with instance_var_set?

Reply:
I can do that! I wasn't sure when I wrote this whether or not exposing `max_queue_size` would be kosher, but I can add it to the initializer and then remove `instance_var_set`.

Reply:
[`2d20be8` (#177)](https://github.com/discourse/prometheus_exporter/pull/177/commits/2d20be835dc00669b3d68b7decb7c24240a974ee)